controllers/vrg_kubeobjects.go

```diff
@@ -106,25 +135,105 @@ func (v *VRGInstance) kubeObjectsProtect(captureNumber int64) []error {
 			objectStore.AddressComponent1(),
 			objectStore.AddressComponent2(),
 			v.s3KeyPrefix(),
-			v.instance.Namespace,
-			VeleroNamespaceNameDefault,
+			sourceNamespacedName,
+			captureInstance,
 			captureNumber,
 		); err != nil {
 			v.log.Error(err, "kube object protect", "profile", s3ProfileName)
-			errors = append(errors, err)
+			// don't return error yet since it could be non-fatal (e.g. slow API server)
+		}
+
+		backupNamespacedName := getBackupNamespacedName(sourceNamespacedName.Name,
+			sourceNamespacedName.Namespace, spec.Name, captureNumber)
+
+		// TODO: remove tight loop here
+		backupComplete := false
+		for !backupComplete {
+			backupComplete, err = backupIsDone(v.ctx, v.reconciler.APIReader,
+				objectWriter{v.reconciler.Client, v.ctx, v.log}, backupNamespacedName)
+
+			if err != nil {
+				backupComplete = true
+			}
 		}
 	}
 
-	return errors
+	return err
 }
 
-func (v *VRGInstance) kubeObjectsRecover(objectStore ObjectStorer) error {
-	if v.instance.Spec.KubeObjectProtection == nil {
-		v.log.Info("Kube object recovery disabled")
```

Comment:
It looks like the test failed because the `spec.kubeObjectProtection == nil` check is missing.  Probably a merge issue.  I'm thinking it just needs to go up a level to before the type group loop.  Feel free to fix this or I can when I merge in the replica store manager changes.
```sh
•••••••••••••••••••••••••••••••••••••••••••••panic: reflect: call of reflect.Value.Set on zero Value

goroutine 1168 [running]:
reflect.flag.mustBeExportedSlow(0x40)
	/opt/hostedtoolcache/go/1.17.12/x64/src/reflect/value.go:237 +0xc5
reflect.flag.mustBeExported(...)
	/opt/hostedtoolcache/go/1.17.12/x64/src/reflect/value.go:231
reflect.Value.Set({0x19a2f00, 0xc00080ca00, 0xc000a04580}, {0x0, 0x0, 0x453fdc})
	/opt/hostedtoolcache/go/1.17.12/x64/src/reflect/value.go:1913 +0x9f
github.com/ramendr/ramen/controllers_test.fakeObjectStorer.DownloadObject({{0xc000e87390, 0xa}, {0xc000820a40, 0x1b}, {0xc0006f7398, 0x6}, 0xc00070eff0}, {0xc000a04580, 0x3e}, {0x1aa1220, ...})
	/home/runner/work/ShioRamen/ShioRamen/controllers/s3utils_test.go:[113](https://github.com/RamenDR/ShioRamen/runs/7569803457?check_suite_focus=true#step:5:114) +0x1e5
github.com/ramendr/ramen/controllers.downloadTypedObject({0x1e21238, 0xc000574000}, {0xc00130ce88, 0x1d}, {0x1dc0110, 0x1}, {0x1aa[122](https://github.com/RamenDR/ShioRamen/runs/7569803457?check_suite_focus=true#step:5:123)0, 0xc00080ca00})
	/home/runner/work/ShioRamen/ShioRamen/controllers/s3utils.go:364 +0x103
github.com/ramendr/ramen/controllers.(*VRGInstance).kubeObjectsRecover(0xc00130db50, {0x1e2[123](https://github.com/RamenDR/ShioRamen/runs/7569803457?check_suite_focus=true#step:5:124)8, 0xc000574000})
	/home/runner/work/ShioRamen/ShioRamen/controllers/vrg_kubeobjects.go:245 +0x14c
github.com/ramendr/ramen/controllers.(*VRGInstance).fetchAndRestorePV(0xc00[130](https://github.com/RamenDR/ShioRamen/runs/7569803457?check_suite_focus=true#step:5:131)db50)
	/home/runner/work/ShioRamen/ShioRamen/controllers/vrg_volrep.go:1533 +0x746
github.com/ramendr/ramen/controllers.(*VRGInstance).restorePVsForVolRep(0xc00130db50)
	/home/runner/work/ShioRamen/ShioRamen/controllers/vrg_volrep.go:1474 +0x11d
github.com/ramendr/ramen/controllers.(*VRGInstance).restorePVs(0xc00130db50)
	/home/runner/work/ShioRamen/ShioRamen/controllers/volumereplicationgroup_controller.go:522 +0x565
github.com/ramendr/ramen/controllers.(*VRGInstance).processAsPrimary(0xc00130db50)
	/home/runner/work/ShioRamen/ShioRamen/controllers/volumereplicationgroup_controller.go:747 +0x3ad
github.com/ramendr/ramen/controllers.(*VRGInstance).processVRGActions(0xc00130db50)
	/home/runner/work/ShioRamen/ShioRamen/controllers/volumereplicationgroup_controller.go:437 +0x20a
github.com/ramendr/ramen/controllers.(*VRGInstance).processVRG(0xc00130db50)
```